src/utils/pgwire/src/pg_protocol.rs

```diff
@@ -59,6 +61,11 @@ static RW_QUERY_LOG_TRUNCATE_LEN: LazyLock<usize> =
         }
     });
 
+tokio::task_local! {
+    /// The current session. Concrete type is erased for different session implementations.
+    pub static CURRENT_SESSION: Weak<dyn Any + Send + Sync>
```

Comment:
Don't really get why we use `Weak` here. Why not just use `Arc`?

Reply:
Good catch. It seems that the session will last until the `PgProtocol` is dropped. I used to believe that there could be multiple sessions, and that a session would end when processing a message. However, `Weak` seems always safer. 🤣